sim/common/vanilla/emerald_dragon_whelp.go

```diff
@@ -0,0 +1,148 @@
+	whelp.registerAcidSpitSpell()
+
+	whelp.EnableAutoAttacks(whelp, core.AutoAttackOptions{
+		// TODO: Need whelp data
+		MainHand: core.Weapon{
+			// These stats are a complete guess from looking at the lone log I could find with Dragon's Call below
+			// https://vanilla.warcraftlogs.com/reports/tQW9mqDrx3R4AdYZ#type=damage-done&ability=-13049&boss=-2&difficulty=0&wipes=2&source=25
+			BaseDamageMin: 80.0,
+			BaseDamageMax: 100.0,
+			SwingSpeed:    1.5,
+			SpellSchool:   core.SpellSchoolPhysical,
+		},
+		AutoSwingMelee: true,
+	})
+
+	return whelp
+}
+
+func emeraldWhelpingStatInheritance() core.PetStatInheritance {
+	return func(ownerStats stats.Stats) stats.Stats {
+		// TODO: Needs more verification
+		return stats.Stats{}
+	}
+}
+
+func (whelp *EmeraldDragonWhelp) Initialize() {}
+
+func (whelp *EmeraldDragonWhelp) ExecuteCustomRotation(sim *core.Simulation) {
+	if !whelp.AcidSpit.IsReady(sim) {
+		whelp.WaitUntil(sim, whelp.AcidSpit.CD.ReadyAt())
+		return
+	}
+
+	whelp.AcidSpit.Cast(sim, whelp.CurrentTarget)
+}
+
+func (whelp *EmeraldDragonWhelp) Reset(sim *core.Simulation) {
+	whelp.Disable(sim)
+}
+
+func (whelp *EmeraldDragonWhelp) OnPetDisable(sim *core.Simulation) {
+}
+
+func (whelp *EmeraldDragonWhelp) GetPet() *core.Pet {
+	return &whelp.Pet
+}
+
+func (whelp *EmeraldDragonWhelp) registerAcidSpitSpell() {
+	actionID := core.ActionID{SpellID: 9591}
+
+	whelp.AcidSpit = whelp.RegisterSpell(core.SpellConfig{
+		ActionID:    actionID,
+		SpellSchool: core.SpellSchoolNature,
+		DefenseType: core.DefenseTypeMagic,
+		ProcMask:    core.ProcMaskSpellDamage,
+		// All of the casts and hits in the above log had the same damage so it would seem debuffs are ignored
+		Flags: core.SpellFlagIgnoreModifiers,
+
+		// This is causing errors because of nil spell.Unit.Env in mana.go:315 so setting a cooldown instead
+		// ManaCost: core.ManaCostOptions{
+		// 	FlatCost: 90,
+		// },
+
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD:      core.GCDDefault,
+				CastTime: time.Second * 3,
+			},
+			CD: core.Cooldown{
+				Timer:    whelp.NewTimer(),
+				Duration: time.Second * 2,
+			},
+			ModifyCast: func(sim *core.Simulation, spell *core.Spell, cast *core.Cast) {
+				castTime := whelp.ApplyCastSpeedForSpell(cast.CastTime, spell)
+
+				if castTime > 0 {
+					whelp.AutoAttacks.StopMeleeUntil(sim, sim.CurrentTime+castTime, false)
+				}
+			},
+		},
+
+		DamageMultiplier: 1,
+		ThreatMultiplier: 1,
+		BonusCoefficient: 1,
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			spell.CalcAndDealDamage(sim, target, sim.Roll(64, 86), spell.OutcomeMagicHitAndCrit)
+		},
+	})
+}
+
+func MakeEmeraldDragonWhelpTriggerAura(agent core.Agent) {
+	character := agent.GetCharacter()
+
+	core.MakeProcTriggerAura(&character.Unit, core.ProcTrigger{
+		ActionID: core.ActionID{SpellID: 13049},
+		Name:     "Emerald Dragon Whelp Proc",
+		Callback: core.CallbackOnSpellHitDealt,
+		ProcMask: core.ProcMaskMelee,
```

Comment:
This would make the trigger also proc on hits with another weapon. Some procs do work like that i'm just wondering if this is one of them.